pkg/runtime/inject/inject_test.go

```
@@ -0,0 +1,193 @@
+			},
+		}
+		client := client.DelegatingClient{}
+		res, err := ClientInto(client, instance)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(res).To(Equal(true))
+		res, err = ClientInto(client, neg)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(res).To(Equal(false))
+	})
+
+	It("should set scheme", func() {
+		instance := testSource{
+			scheme: func(c *runtime.Scheme) error {
+				if c != nil {
+					return nil
+				}
+				return fmt.Errorf("")
+			},
+		}
+		scheme := runtime.NewScheme()
+		res, err := SchemeInto(scheme, instance)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(res).To(Equal(true))
+		res, err = SchemeInto(scheme, neg)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(res).To(Equal(false))
+	})
+
+	It("should set stop channel", func() {
+		instance := testSource{
+			stop: func(c <-chan struct{}) error {
+				if c != nil {
+					return nil
+				}
+				return fmt.Errorf("")
+			},
+		}
+		stop := make(chan struct{})
+		res, err := StopChannelInto(stop, instance)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(res).To(Equal(true))
+		res, err = StopChannelInto(stop, neg)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(res).To(Equal(false))
+	})
+
+	It("should set stop channel", func() {
+		instance := testSource{
+			f: func(c Func) error {
+				if c != nil {
+					return nil
+				}
+				return fmt.Errorf("")
+			},
+		}
+		f := func(interface{}) error { return nil }
+		res, err := InjectorInto(f, instance)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(res).To(Equal(true))
+		res, err = InjectorInto(f, neg)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(res).To(Equal(false))
+	})
+
+})
+
+type testSource struct {
+	scheme func(scheme *runtime.Scheme) error
+	cache  func(cache.Cache) error
+	config func(config *rest.Config) error
+	client func(client.Client) error
+	f      func(Func) error
+	stop   func(<-chan struct{}) error
+}
+
+func (s testSource) InjectCache(c cache.Cache) error {
+	if s.cache != nil {
+		return s.cache(c)
+	}
+	return nil
```

This should probably `Fail("unexpected call")` or something right?